tests/test_connection.py

```diff
@@ -259,3 +259,63 @@ def test_temporary_connection(mock_conn, option, runner):
         schema="PUBLIC",
         warehouse="xsmall",
     )
+
+
+@mock.patch("snowflake.connector.connect")
+@pytest.mark.parametrize("option", ["--temporary-connection", "-x"])
+def test_key_pair_authentication(mock_conn, option, runner):
```

Comment:
This test works but tests in `test_common_decorators` started to fail.
https://github.com/Snowflake-Labs/snowcli/actions/runs/6508833780/job/17737398464?pr=478